qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/FeedbackActivity.kt

```diff
@@ -135,6 +129,30 @@ class FeedbackActivity : AppCompatActivity() {
             enableButtonStoreFeedbackMetadata(feedbackMetadataWrapper != null && value != null)
         }
 
+    private val mapboxNavigation by requireMapboxNavigation(
+        onCreatedObserver = object : MapboxNavigationObserver {
+            override fun onAttached(mapboxNavigation: MapboxNavigation) {
+                binding.mapView.location.apply {
+                    setLocationProvider(navigationLocationProvider)
+                    enabled = true
+                }
+                mapboxNavigation.registerLocationObserver(locationObserver)
+                mapboxNavigation.registerRoutesObserver(routesObserver)
+            }
+
+            override fun onDetached(mapboxNavigation: MapboxNavigation) {
+                mapboxNavigation.unregisterLocationObserver(locationObserver)
+                mapboxNavigation.unregisterRoutesObserver(routesObserver)
+            }
+        }
+    ) {
+        MapboxNavigationApp.setup {
+            NavigationOptions.Builder(this)
+                .accessToken(Utils.getMapboxAccessToken(this))
+                .build()
+        }
```

Comment:
Here you have the same problem as in mapbox/mapbox-navigation-android-examples#129. 
When you close one example and go to another one, its navigation options will be ignored. 
Also activity from the first example will be leaked via navigation options stored in `MapboxNavigationApp`.

Reply:
Yeah we need a disable in the orchestrating activity

> Also activity from the first example will be leaked via navigation options stored in MapboxNavigationApp.

Why will this be leaked? The `NavigationOptions` holds onto the [applicationContext](https://github.com/mapbox/mapbox-navigation-android/blob/9073234c2978c019980ca606b16fa0d3ff37f85b/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt#L176) not the Activity Context

Reply:
`NavigationOptionsProvider` instance you create here references the activity, that's why it is leaking. This is also reported by LeakCanary. Looks like you should use another `MapboxNavigationApp.setup` overload:
```suggestion
        MapboxNavigationApp.setup(
            NavigationOptions.Builder(this)
                .accessToken(Utils.getMapboxAccessToken(this))
                .build(),
        )
```

Reply:
Yeah that totally makes sense, thanks! That is terrible